addon-test-support/@ember/test-helpers/setup-rendering-context.ts

```diff
@@ -144,6 +144,8 @@ export function render(template: TemplateFactory): Promise<void> {
     };
     toplevelView.setOutletState(outletState);
 
+    resetScrollState();
```

Comment:
I'm not sure if this is the correct place to call this. It would still have the same issue if a rendering test scrolls somewhere and then an application test follows that relies on the scroll position. Should we call this in "`afterEach()`" instead? 🤔

Reply:
I think I understand that issue. Please help me understand your thoughts on the recommendation for `afterEach()`. 

Are you suggestion that perhaps `resetScrollState()` should be a helper and then it'd be at the discretion of the user to call it in the afterEach hook whenever scroll state needs to be guaranteed? That way it will work regardless if it is an acceptance test or an integration test.

```
  setupRenderingTest(hooks);
 //OR
 //setupApplicationTest(hooks);

  hooks.afterEach(async function() {
    await resetScrollState()
  });

  test('it renders', async function(assert) {
    await render(hbs`<MyComponent />`);
});

```

Reply:
hmmm. But even that wouldn't solve the issue is there were two renders in a single component test.

```
    await render(hbs`
      await render(hbs`<MyComponent />`);
    `);

    clearRender();

    await render(hbs`
      <div>SomethingElse</div>
    `);

```

So maybe it has to be in either `render()` or `clearRender()` for integration tests as well as used in `afterEach` for acceptance tests?

Reply:
> But even that wouldn't solve the issue is there were two renders in a single component test.

FWIW, I think this is pretty rare.

---

We can add logic to `teardownContext` and `clearRender` to reset the scroll state, that would work for both rendering and application tests as well as using `render` twice...